src/clib/pio_gdal.c

```diff
@@ -0,0 +1,1350 @@
+	  size_t type_size;
+
+	  var_ndims = 1; // FIXED FOR NOW. For data-read purposes, it's a 1D stream across the number of
+                         // elements.
+	  (*ndims)[v] = var_ndims;
+	  OGRFieldType Fld = OGR_Fld_GetType(OGR_FD_GetFieldDefn(hFD,v));
+	  bool typeOK = true; // assume we're good
+	  switch (Fld) {
+	  case OFTReal:
+	    (*pio_type)[v] = (int)PIO_DOUBLE;
+	    (*pio_type_size)[v] = sizeof(double);
+	    break;
+	  case OFTInteger:
+	    (*pio_type)[v] = (int)PIO_INT;
+	    (*pio_type_size)[v] = sizeof(int);
+	    break;
+	  case OFTString:
+	    (*pio_type)[v] = (int)PIO_STRING;
+	    (*pio_type_size)[v] = -1;
+	    break;
+	  // This needs to be done. How do we deal with timestamps etc in GDAL vector fields?
+	  //>>case OFTDate:
+	  //>>  break;
+	  //>>	    case OFTTime:
+	  //>>	      break;
+	  //>>	    case OFTDate:
+	  //>>	      break;
+	  //>>	    case OFTDateTime:
+	  default:
+	    typeOK = false;
+	    break;
+	  }
+	  if (!typeOK) // Not a usable type 
+	    continue;
+
+	  /* Get the MPI type corresponding with the PIO type. */
+	  if ((ret = find_mpi_type((*pio_type)[v], &(*mpi_type)[v], NULL)))
+	    return pio_err(NULL, file, ret, __FILE__, __LINE__);
+	  
+	  /* Get the size of the MPI type. */
+	  if ((*mpi_type)[v] == MPI_DATATYPE_NULL)
+	    (*mpi_type_size)[v] = 0;
+	  else
+	    if ((mpierr = MPI_Type_size((*mpi_type)[v], &(*mpi_type_size)[v])))
+	      return check_mpi(NULL, file, mpierr, __FILE__, __LINE__);
+	  
+	} /* next var */
+    } /* If PIO_TYPE_GDAL */
+    return PIO_NOERR;
+}
+
+/**
+ * The PIO-C interface for the GDAL function OGR_L_FindFieldIndex()
+ *
+ * This routine is called collectively by all tasks in the communicator
+ * ios.union_comm.
+ *
+ * @param ncid the ncid of the open file, obtained from
+ * GDALc_openfile() or GDALc_createfile().
+ * @param name the field name.
+ * @param varidp a pointer that will get the variable id
+ * @return PIO_NOERR for success, error code otherwise.  <<See GDALc_Set_File_Error_Handling>>
+ * @ingroup PIO_inq_var_c
+ * @author Michael Long (adapted from Jim Edwards, Ed Hartnett)
+ */
+int
+GDALc_inq_fieldid(int fileid, const char *name, int *fieldidp)
+{
+    iosystem_desc_t *ios;  /* Pointer to io system information. */
+    file_desc_t *file;     /* Pointer to file information. */
+    int ierr;              /* Return code from function calls. */
+    int mpierr = MPI_SUCCESS, mpierr2;  /* Return code from MPI function codes. */
+
+    /* Get file info based on fileid. */
+    if ((ierr = pio_get_file(fileid, &file)))
+        return pio_err(NULL, NULL, ierr, __FILE__, __LINE__);
+    ios = file->iosystem;
+
+    /* Caller must provide name. */
+    if (!name || strlen(name) > NC_MAX_NAME)
+        return pio_err(ios, file, PIO_EINVAL, __FILE__, __LINE__);
+
+    PLOG((1, "GDALc_inq_fieldid fileid = %d name = %s", fileid, name));
+
+    if (ios->async)
+    {
+        if (!ios->ioproc)
+        {
+            int msg = PIO_MSG_INQ_VARID;
```

Comment:
You will need to define unique msg id's for your functions, this is part of the async support.